apps/site/src/app/portal/Portal.tsx

```diff
@@ -1,21 +1,62 @@
-import api from "@/lib/utils/api";
+"use client";
 
-async function getIdentity(): Promise<string> {
-	const res = await api.get<string>("/user/me");
-	return res.data;
+import Head from "next/head";
+import { useRouter } from "next/navigation";
+import { useEffect } from "react";
+
+import VerticalTimeline from "./VerticalTimeline";
+import Message from "./Message";
+import { Identity } from "@/lib/utils/getUserIdentity";
+
+import styles from "./Portal.module.scss";
+
+export const enum PortalStatus {
+	pending = "PENDING_REVIEW",
+	reviewed = "REVIEWED",
+	accepted = "ACCEPTED",
+	rejected = "REJECTED",
+	waitlisted = "WAITLISTED",
+	confirmed = "CONFIRMED",
+}
+
+interface PortalProps {
+	identity: Identity;
 }
 
-async function Portal() {
-	const identity = await getIdentity();
+function Portal({ identity }: PortalProps) {
+	const router = useRouter();
+	const status = identity.status;
+
+	useEffect(() => {
+		if (status === null) {
+			router.push("/apply");
+		}
+	}, [status, router]);
+
+	if (status === null) {
+		return null;
+	}
+
 	return (
-		<div className="min-h-screen flex flex-col justify-center bg-white text-black">
-			<h1>Hello</h1>
-			{Object.entries(identity).map(([key, value]) => (
-				<p key={key}>
-					{key} - {value}
-				</p>
-			))}
-		</div>
+		<>
+			<section className=" w-full flex items-center flex-col">
+				<div className="m-24">
+					<Head>
+						<title>Portal | IrvineHacks 2024</title>
+					</Head>
```

Comment:
Nitpick: Instead of having the `title` here, you can [ `export const metadata`](https://nextjs.org/docs/app/building-your-application/optimizing/metadata) in `page.tsx` for `portal` like how it's done in [`layout.tsx`](https://github.com/HackAtUCI/irvinehacks-site-2024/blob/main/apps/site/src/app/layout.tsx).